src/tir/schedule/concrete_schedule.cc

```diff
@@ -452,6 +452,9 @@ Array<LoopRV> ConcreteScheduleNode::Split(const LoopRV& loop_rv,
       if (is_const_int(factor) && !is_positive_const(factor)) {
         throw NonPositiveFactorError(state_->mod, factor.as<IntImmNode>()->value, i);
       }
+      if (factor.dtype().bits() > loop->extent.dtype().bits()) {
+        factor = IntImm(loop->extent.dtype(), factor.as<IntImmNode>()->value);
+      }
```

Comment:
A `factor` isn't necessary a constant integer, so casting it using `factor.as<IntImmNode>()` is probably not the best here. Instead please use:

```python
      if (factor.dtype().bits() > loop->extent.dtype().bits()) {
        factor = tir::cast(loop->extent.dtype(), factor);
      }
```